com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -246,6 +247,11 @@ private boolean serviceHasDifferentTypeOrLabel(CloudServiceExtended service, Clo
         return haveDifferentTypes || haveDifferentLabels;
     }
 
+    private boolean hasServiceDeleteFailedState(ServiceOperation lastOperation) {
```

Comment:
This method is almost exactly the same as hasServiceCreateFailedState.
Refactor them to reuse the code and remove the duplication.

Reply:
done